wallet/wallet-rpc-client/src/rpc_client/mod.rs

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) 2023 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+pub mod client_impl;
+
+use rpc::new_http_client;
+use rpc::RpcAuthData;
+use rpc::RpcHttpClient;
+
+use crate::wallet_rpc_traits::WalletInterface;
+
+#[derive(thiserror::Error, Debug)]
+pub enum WalletRpcError {
+    #[error("Initialization error: {0}")]
+    InitializationError(Box<WalletRpcError>),
+    #[error("Decoding error: {0}")]
+    DecodingError(#[from] serialization::hex::HexError),
+    #[error("Client creation error: {0}")]
+    ClientCreationError(jsonrpsee::core::ClientError),
+    #[error("Response error: {0}")]
+    ResponseError(jsonrpsee::core::ClientError),
```

Comment:
Rather than using jsonrpsee directly here (and adding codecheck exception), re-export the type from the `rpc` crate and use that.